Makefile

```diff
@@ -124,7 +124,7 @@ DEFVCPUS := 1
 # Default maximum number of vCPUs
 DEFMAXVCPUS := 0
 # Default memory size in MiB
-DEFMEMSZ := 2048
+DEFMEMSZ := 128
```

Comment:
I don't really get what the `predefined memory` is.
If `docker run` without memory limit, we will have a 128MB vm, which is assigned when qemu setup.
If `docker run --memory 100M`, we will have a 128MB+100MB vm, the 100MB is hotplug memory.
What else do we need?

Reply:
@clarecch yes, for your use case `docker run` without limited memory the memory that the container will have  is just 128MB?. I feel that just 128MB for a not limited container is a little of memory, dont you think?

I mean it is great for density starting all the not limited containers with 128MB, but at the same time not sure if is a good default for most of the container applications running today.

Reply:
Maybe we should add a new config field name `SYSRESMEMSZ(system reserve memory)` which default 128MB. And keeping `DEFMEMSZ` to be 2GB. If memory.limit is not defined, start with `DEFMEMSZ`, or else start with `SYSRESMEMSZ` + memory.limit.

Reply:
@jcvenegas When we create a pod, we want to start the vm with the minimal memory, but the `pause container` has no memory limit defined, which means each pod will have 1 or 2GB base memory size. That's why I choose 128MB before. @sboeuf @bergwolf  @jodh-intel WDYT

Reply:
@jodh-intel asked me a question, which raised a point we should make, and another one that springs to mind....

1) note that, although we (currently) hand the VM 2G of memory *space* - that does not mean it allocates and consumes 2G of host side memory, unless it is used. Sure, once you start doing 'interesting stuff' in your container it will probably consume it all by filling it with cache - but, it should be noted that handing the VM more or less memory at startup does not carry a large host side overhead, at least initially afaik.

2) Generally we try to be as compatible with the other container runtimes (meaning pretty much `runc`) as we can. If you run a runc container or pod with no memory restrictions, then I believe you get 'all of host memory'. Restricting us to 128M will be the opposite extreme of that. I'm not sure that is a good thing - I'm pretty sure we will start getting a lot of reports of 'I ran my mysql DB and it fell over'.

We can't really 'win' here - if we hand the VM 128M by default, many workloads will fail under the 'default memory' situation. If we hand the VM 'all of host memory', then on large systems that does carry a lot of overhead for the VM to manage it, and then we will consume host memory pretty quickly, and launching more VMs will be refused (as by default the host kernels check that the RAM is available before they allow the VM to launch - they do a pre-overcommit check).

We have had a 2G default for a couple of years now (across clear containers and now kata), and have run into remarkably few situations where anybody has found it a problem. Do we have any prior experience to add here from the runv side (@gnawux maybe?) - maybe that can help guide us here.

128M worries me a little as it is pretty small. That would probably be fine if we knew all workloads we ever get have a defined memory requirement/resource/constraint - but, afaik, they don't (or it is not a guarantee). And, I don't think we know that in advance, so we can't even dynamically choose depending on the workload can we?

Reply:
Also, see: https://github.com/kata-containers/runtime/blob/master/cli/config.go#L693 - we will fail if the memory level is "too low".